js/src/credentialRequest/credentialRequestFactory.js

```diff
@@ -0,0 +1,20 @@
+const LdpVcCredentialRequest = require("./types/LdpVcCredentialRequest");
+const {CredentialFormat} = require("../constants/CredentialFormat");
+
+class CredentialRequestFactory {
+   static createCredentialRequest(
+    credentialFormat,
+    accessToken,
+    issuerMetaData,
+    proof
+  ) {
+    switch (credentialFormat) {
+      case CredentialFormat.LDP_VC:
+        return new LdpVcCredentialRequest(accessToken, issuerMetaData, proof).constructRequest();
+      default:
+        throw new Error("Unsupported credential format");
```

Comment:
Can we extract UnsupportedCredentialFormatException or use available DownloadFailedException with additional error info as Unsupported credential format